### PR TITLE
Update the search-query-syntax examples to show multiple example search terms

### PR DESCRIPTION
**Pull request by michael-berk**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/michael-berk/mlflow/pull/11191?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11191/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11191
```

</p>
</details>

### Related Issues/PRs
https://databricks.atlassian.net/browse/ML-38533

### What changes are proposed in this pull request?
* Redo search experiments

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [x] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/search-runs.rst [hunk 1]
```diff
@@ -93,139 +93,183 @@ Then paste the following code to a local pythnon file, e.g., ``~/create_testing_
...
-MLflow.
+To filter your MLflow runs, you need to write **search queries**, which are SQL-like conditions. 
 
-.. code-block:: sql
+To showcase this functionality, let's reopen the MLflow UI, let's explore some example fields that you can use to filter your runs in the MLflow UI.
+
+.. figure:: _static/images/search-runs/before_search_query.png
+   :alt: search query
+   :width: 90%
+   :align: center
```
Comment (harupy) [inline docs/source/search-runs.rst:120]: Can we use code example instead a screenshot? Screenshots can be outdated easily.

File: docs/source/search-runs.rst [hunk 2]
```diff
@@ -1,45 +1,46 @@
...
-MLflow UI provides a search bar to search your MLflow runs, which can be found in the experiment view, as shown
-by the screenshot:
-
-.. figure:: _static/images/search-runs/run_search_bar.png
-   :alt: Run search bar
-   :width: 90%
-   :align: center
-
+The MLflow UI provides a powerful search interface that allows you to filter runs. Below we'll...
+1. Create example mllfow runs
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:13]:
```suggestion
1. Create example MLflow runs
```

File: docs/source/search-runs.rst [hunk 3]
```diff
@@ -1,45 +1,46 @@
...
-First let's start a local MLflow UI service by running:
+Before running the script, let's simply start the MLflow UI on a local host. 
 
 .. code-block:: bash
 
   mlflow ui
 
-Then paste the following code to a local pythnon file, e.g., ``~/create_testing_runs.py``, and execute it.
+Let's visit ``http://localhost:5000/`` in our web browser. After doing so, you'll notice that we don't 
+have any experiments or models. Let's resolve this by creating a few MLflow runs. Via the script
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:32]:
```suggestion
have any experiments or models. Let's resolve this by creating a few MLflow runs via the script
```

File: docs/source/search-runs.rst [hunk 4]
```diff
@@ -1,45 +1,46 @@
...
+below. 
+
+Note that when you run this script, you'll want to do so from the same directory that your ran
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:35]:
```suggestion
Note that when you run this script, you'll want to do so from the same directory that you ran
```

File: docs/source/search-runs.rst [hunk 5]
```diff
@@ -93,139 +93,183 @@ Then paste the following code to a local pythnon file, e.g., ``~/create_testing_
           mlflow.log_input(dataset, context=dataset_context[i])
 
 
-The code above creates 10 MLflow runs with different metrics, params, tags and dataset information. After execution,
-you should find all these runs under the experiment "/search-run-guide", as shown by the screenshot:
+The code above creates 10 MLflow runs with different metrics, params, tags and dataset information. 
+After successful execution, if you return to the MLflow UI in your browser, you should find all
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:97]:
```suggestion
After successful execution, if you return to the MLflow UI in your browser, you should find all of 
```

File: docs/source/search-runs.rst [hunk 6]
```diff
@@ -93,139 +93,183 @@ Then paste the following code to a local pythnon file, e.g., ``~/create_testing_
           mlflow.log_input(dataset, context=dataset_context[i])
 
 
-The code above creates 10 MLflow runs with different metrics, params, tags and dataset information. After execution,
-you should find all these runs under the experiment "/search-run-guide", as shown by the screenshot:
+The code above creates 10 MLflow runs with different metrics, params, tags and dataset information. 
+After successful execution, if you return to the MLflow UI in your browser, you should find all 
+these runs under the experiment "search-run-guide", as shown by the screenshot:
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:98]:
```suggestion
these runs under the experiment "search-run-guide", as shown by the following screenshot:
```

File: docs/source/search-runs.rst [hunk 7]
```diff
@@ -93,139 +93,183 @@ Then paste the following code to a local pythnon file, e.g., ``~/create_testing_
...
 
 .. figure:: _static/images/search-runs/created_mlflow_runs.png
    :alt: testing runs
    :width: 90%
    :align: center
 
+Now in production systems, it's common to have thousands or even hundreds of thousands of mlflow
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:105]:
```suggestion
In real-world production deployments of MLflow, it's common to have thousands or even hundreds of thousands of  
```

File: docs/source/search-runs.rst [hunk 8]
```diff
@@ -93,139 +93,183 @@ Then paste the following code to a local pythnon file, e.g., ``~/create_testing_
...
+^^^^^^^^^^^^^^^^^^^^
 
-You need to write search queries to fetch your MLflow runs. The query must be a condition or a few conditions
-concatenated by ``and``. At a high level, the syntax is similar to SQL WHERE clause without supporting ``or`` keyword.
-For example, below is a query to search for all runs that have logged a ``loss`` metric smaller than ``1`` to
-MLflow.
+To filter your MLflow runs, you need to write **search queries**, which are SQL-like conditions. 
 
-.. code-block:: sql
+To showcase this functionality, let's reopen the MLflow UI, let's explore some example fields that you can use to filter your runs in the MLflow UI.
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:115]:
```suggestion
To showcase this functionality, let's reopen the MLflow UI and explore some example fields that you can use to filter your runs.
```

File: docs/source/search-runs.rst (hunk 1 above)
Comment (BenWilson2) [reply docs/source/search-runs.rst:120]: +1

File: docs/source/search-runs.rst [hunk 9]
```diff
@@ -93,139 +93,183 @@ Then paste the following code to a local pythnon file, e.g., ``~/create_testing_
...
 
-.. note::
-  There is a caveat that using ``IN`` requires wrapping string with single quotes, e.g.,
-  ``params."learning rate" IN ('0.001', '0.01')``. ``params."learning_rate" IN ("0.001", "0.01")`` will throw an error.
+As noted above, MLflow search syntax is similar to SQL with a few notable exceptions.
 
-Handling Special Characters in MLflow Field
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+* The SQL ``OR`` keyword is not supported.
+* For fields that contain special characters or start with numbers, you need to wrap them in double quotes.
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:139]: Let's make this a callout as well above line 138. This is quite possibly the most frequently overlooked component of this search syntax.

File: docs/source/search-runs.rst [hunk 10]
```diff
@@ -93,139 +93,183 @@ Then paste the following code to a local pythnon file, e.g., ``~/create_testing_
...
-cases, you will need to wrap it by double quotes. The same rule applies when your field starts with numbers.
-For example if you ever logged something by ``mlflow.log_metric("cross-entropy-loss", 0.15)``, then your query will be
-similar to:
+  * Good: ``metrics."cross-entropy-loss" < 0.5``
+  * Bad:  ``metrics.cross-entropy-loss < 0.5``
+  * Good: ``params."1st_iteration_timestamp" = "2022-01-01"``
+  * Bad:  ``params.1st_iteration_timestamp = "2022-01-01"``
 
-.. code-block:: sql
+* For the SQL ``IN`` keyword, you must surround the values of your list with single quotes.
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:146]: For the good / bad examples, can we make it more visually apparent the difference between them?

```rst
.. code-block:: diff

    - Bad: params."learning rate" IN ("0.001", "0.01")
    + Good: params."learning rate" IN ('0.001', '0.01')
```
(might require some adjustment, but is likely to be very obvious at a quick glance which one to pay attention to for a quick reference)

File: docs/source/search-runs.rst [hunk 11]
```diff
@@ -93,139 +93,183 @@ Then paste the following code to a local pythnon file, e.g., ``~/create_testing_
...
+
+  * Good: ``params."learning rate" IN ('0.001', '0.01')``
+  * Bad:  ``params."learning rate" IN ("0.001", "0.01")``
+
+* For the SQL ``IN`` keyword, you can only search the following fields:
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:151]: This could use some refinement. Parameters can be searched, right? 

For the datasets.{any_attribute} condition, can an example be provided?

File: docs/source/search-runs.rst [hunk 12]
```diff
@@ -93,139 +93,183 @@ Then paste the following code to a local pythnon file, e.g., ``~/create_testing_
...
+* Non-None conditions for numeric fields are not supported e.g. ``metrics.accuracy != "None"`` will fail.
+
+Other than the that, the syntax should be intuitive to anyone who has used SQL. To assemble
+a single search condition, you must assemble an inequality using the following components...
+
+1. **An MLflow field**: a metric, param, tag, dataset or run metadata.
+2. **A comparator**: an inequality operator. 
+
+  * For numerics, MLflow supports ``=``, ``!=``, ``>``, ``>=``, ``<``, and ``<=``.
+  * For strings, MLflow supports ``=``, ``!=``, ``LIKE`` (case-sensitive) and ``ILIKE`` (case-insensitive).
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:165]: Let's provide some examples of common query terms (particularly LIKE and ILIKE) and some range search conditions (metric_A >= 0.7654, etc.)

File: docs/source/search-runs.rst [hunk 13]
```diff
@@ -93,139 +93,183 @@ Then paste the following code to a local pythnon file, e.g., ``~/create_testing_
...
+
+Other than the that, the syntax should be intuitive to anyone who has used SQL. To assemble
+a single search condition, you must assemble an inequality using the following components...
+
+1. **An MLflow field**: a metric, param, tag, dataset or run metadata.
+2. **A comparator**: an inequality operator. 
+
+  * For numerics, MLflow supports ``=``, ``!=``, ``>``, ``>=``, ``<``, and ``<=``.
+  * For strings, MLflow supports ``=``, ``!=``, ``LIKE`` (case-sensitive) and ``ILIKE`` (case-insensitive). 
+  * For sets, MLflow supports ``IN``.
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:166]: Let's add an example of searching over a set

File: docs/source/search-runs.rst [hunk 14]
```diff
@@ -93,139 +93,183 @@ Then paste the following code to a local pythnon file, e.g., ``~/create_testing_
...
+side of the inequality. Note that they are **stored as numbers**, so you must use numeric comparators.
 
 .. code-block:: sql
 
-    metrics.accuracy > 0.72
-    metrics.loss <= 0.15
-    metrics."log-scale-loss" <= 0
-    metrics."f1 score" >= 0.5
-    metrics.accuracy > 0.72 AND metrics.loss <= 0.15
+  metrics.accuracy > 0.72
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:192]: is metrics."accuracy" > 0.72 valid? Let's show that both are options.

File: docs/source/search-runs.rst [hunk 15]
```diff
@@ -93,139 +93,183 @@ Then paste the following code to a local pythnon file, e.g., ``~/create_testing_
...
+
+Note that dataset information is **stored as strings**, so you must use string comparators, such as ``=`` and ``!=``.
+Also note that datasets support set comparators, such as ``IN``.
 
 .. code-block:: sql
 
     datasets.name like "custom"
     datasets.digest IN ('s8ds293b', 'jks834s2')
-    datasets.context == "train"
+    datasets.context = "train"
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:257]: For line 255 above, can we fix the DML statement to be capitalized? While we do support "like", standard ANSI SQL lint formatting dictates that DML operators are all CAPS

File: docs/source/search-runs.rst [hunk 16]
```diff
@@ -93,139 +93,183 @@ Then paste the following code to a local pythnon file, e.g., ``~/create_testing_
...
+
+5 - Searching By Run's Metadata
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Run metadata are a variety of user-specified and system-generated attributes that provide additional context about the run.
 
-Searching By Run's Metadata
-~~~~~~~~~~~~~~~~~~~~~~~~~~~
+To search for runs by filtering on run's metadata, you must include the ``attributes`` prefix in the left
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:264]:
```suggestion
To search for runs by filtering on the metadata of runs, you must include the ``attributes`` prefix in the left
```

File: docs/source/search-runs.rst [hunk 17]
```diff
@@ -93,139 +93,183 @@ Then paste the following code to a local pythnon file, e.g., ``~/create_testing_
...
+side of the inequality. Note that run metadata can be either a string or a numeric depending on the 
+attribute, so you must use the appropriate comparator. For a complete list of attributes, see
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:266]: Can we add examples of this?

File: docs/source/search-runs.rst [hunk 18]
```diff
@@ -239,82 +283,168 @@ you need to customize the compared value to match your MLflow runs, as some meta
...
-  )[0]
+  )
+
 
+2 - Ordering
+~~~~~~~~~~~~
 
-To get all active runs from experiments IDs 3, 4, and 17 that used a CNN model
-with 10 layers and had a prediction accuracy of 94.5% or higher, use:
+Another useful feature is ordering the results. You specify a list of columns of interest along with
```
Comment (BenWilson2) [inline docs/source/search-runs.rst:395]:
```suggestion
Another useful feature that is available in the search API is allowing for ordering of the returned search results. You can specify a list of columns of interest along with
```

File: docs/source/search-runs.rst (hunk 11 above)
Comment (michael-berk) [reply docs/source/search-runs.rst:151]: Good catch. Parameters **cannot** be searched with IN

**Review by harupy — APPROVED** [review]
LGTM!
